frontend/pages/hosts/details/cards/HostSoftwareLibrary/HostInstallerActionCell/HostInstallerActionCell.tsx

```diff
@@ -0,0 +1,236 @@
+/**
+ * HostInstallerActionCell is only used on Host details > Software Library card.
+ * Displays install/uninstall buttons for host installers.
+ * HostInstallerActionButton is reused for install/uninstall buttons
+ * in Fleet Desktop > Self-service but with different disabled states and tooltips.
+ */
+
+import React, { useState, useEffect } from "react";
+import Button from "components/buttons/Button";
+import Icon from "components/Icon";
+import TooltipWrapper from "components/TooltipWrapper";
+
+import {
+  IHostSoftwarePackage,
+  IHostAppStoreApp,
+  SoftwareInstallStatus,
+  IHostSoftwareWithUiStatus,
+} from "interfaces/software";
+import { IconNames } from "components/icons";
+import {
+  getInstallerActionButtonConfig,
+  IButtonDisplayConfig,
+} from "../../Software/helpers";
+
+interface IActionButtonState {
+  installDisabled: boolean;
+  installTooltip?: React.ReactNode;
+  uninstallDisabled: boolean;
+  uninstallTooltip?: React.ReactNode;
+}
+
+export interface IGetActionButtonStateProps {
+  hostScriptsEnabled: boolean;
+  softwareId: number;
+  status: SoftwareInstallStatus | null;
+  softwarePackage: IHostSoftwarePackage | null;
+  appStoreApp: IHostAppStoreApp | null;
+  hostMDMEnrolled?: boolean;
+  isMyDevicePage?: boolean;
+}
+
+interface IHostInstallerActionButtonProps {
+  baseClass: string;
+  tooltip?: React.ReactNode;
+  disabled: boolean;
+  onClick: () => void;
+  icon: IconNames;
+  text: string;
+  testId?: string;
+}
+
+interface IHostInstallerActionCellProps {
+  software: IHostSoftwareWithUiStatus;
+  onClickInstallAction: (softwareId: number) => void;
+  onClickUninstallAction: () => void;
+  baseClass: string;
+  hostScriptsEnabled?: boolean;
+  hostMDMEnrolled?: boolean;
+  /** Different disabled states and tooltips compared to host details > library and fleet desktop > self-service */
+  isMyDevicePage?: boolean;
+}
+
+export const getActionButtonState = ({
+  hostScriptsEnabled,
+  status,
+  appStoreApp,
+  hostMDMEnrolled,
+  isMyDevicePage,
+}: IGetActionButtonStateProps): IActionButtonState => {
+  const pendingStatuses = ["pending_install", "pending_uninstall"];
+  let installDisabled = false;
+  let uninstallDisabled = false;
+  let installTooltip: React.ReactNode | undefined;
+  let uninstallTooltip: React.ReactNode | undefined;
```

Comment:
small nit: React.ReactNode includes undefined so we can drop the `| undefined` part.